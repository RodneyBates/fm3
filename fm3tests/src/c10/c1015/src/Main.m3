MODULE Main

(* Test isolated expresssion operators. *)

; CONST A = B OR C
; CONST B = D AND E
; CONST C = NOT F

; CONST D = G = H 
; CONST E = I # J 
; CONST F = K < L 
; CONST G = M <= N 
; CONST H = O > P 
; CONST I = Q >= R
; CONST J = S IN T

; CONST K = A + B
; CONST L = C - D
; CONST M = E & F

; CONST N = G * H 
; CONST O = I / J
; CONST P = K DIV L 
; CONST Q = M MOD N

; CONST R = + O
; CONST S = - P

; CONST T = Q 

; BEGIN
  END Main
.
