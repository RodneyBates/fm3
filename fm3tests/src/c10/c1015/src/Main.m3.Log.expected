[32mINFO:  [0mMain.m3 Pass 1 output file Main.m3.FM3Pass1 has 1410 bytes.
[31mERROR: [0mMain.m3[5,8] Illegal recursive declarations: "A", also involving
       "B" [6,8]
       "C" [7,8]
       "D" [9,8]
       "E" [10,8]
       "F" [11,8]
       "G" [12,8]
       "H" [13,8]
       "I" [14,8]
       "J" [15,8]
       "K" [17,8]
       "L" [18,8]
       "M" [19,8]
       "N" [21,8]
       "O" [22,8]
       "P" [23,8]
       "S" [27,8]
       "Q" [24,8]
       "R" [26,8]
[32mINFO:  [0mMain.m3 Patch stack Main.m3.FM3Patch peak size = 39 bytes.
[32mINFO:  [0mMain.m3 Pass 2 output file Main.m3.FM3Pass2 has 750 bytes.
